iree/turbine/kernel/wave/templates/evoformer.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import iree.turbine.kernel.lang as tkl
+import iree.turbine.kernel.wave as tkw
+from iree.turbine.kernel.lang.global_symbols import *
+from iree.turbine.kernel.wave.constraints import MMAType
+from iree.turbine.kernel._support.dtype import DataType
+from iree.turbine.kernel.wave.utils import (
+    get_mfma_load_elems_per_thread,
+    get_mfma_store_elems_per_thread,
+)
+
+
+def get_evoformer_kernel(
+    batch: tuple[int, int],
+    n: tuple[int, int],
+    kv_seq_len: tuple[int, int],
+    heads: tuple[int, int],
+    head_dim: tuple[int, int],
+    q_seq_len: tuple[int, int],
+    v_dim: tuple[int, int],
+    mfma_variant: MMAType,
```

Comment:
Do we want to expose `mfma_variant` as parameter? As it directly depends on `datatype` and sizes.

Reply:
Right now since we haven't tuned these yet, I would like to keep this here. Once we know which variant works best for which shape we can remove this as a parameter.